Cython/Tests/TestTestUtils.py

```diff
@@ -42,7 +42,7 @@ def test_write_file_dedent(self):
         And no one can talk to a horse
         of course
         """
-        self._test_write_file(text, textwrap.dedent(text).encode('utf8'), dedent=True)
+        self._test_write_file(text, textwrap.dedent(text).encode('utf-8'), dedent=True)
```

Comment:
Fails on Windows: `AssertionError: b'\r\nA horse is a horse,\r\nof course, of c[53 chars]\r\n' != b'\nA horse is a horse,\nof course, of cours[43 chars]se\n' : ("b'\\nA horse is a horse,\\nof course, of course,\\nAnd no one can talk to a horse\\nof course\\n'", "b'\\r\\nA horse is a horse,\\r\\nof course, of course,\\r\\nAnd no one can talk to a horse\\r\\nof course\\r\\n'")`

The reason for this is the different handling of [linesep](https://docs.python.org/3/library/os.html#os.linesep) when writing to files. Since `_test_write_file` uses `'rb'`, the difference is revealed. For the same reason, the `self._test_write_file("\n", "\n".encode('utf-8'))` code will not work, unlike the `self._test_write_file("\n".encode('utf-8'), "\n".encode('utf-8'))
`.

Reply:
Ah, lovely. Yeah, that's worth fixing then. We can probably just do the usual
`.replace('\r\n', '\n').replace('\r', '\n')`
after reading the file back in. It's test code, that'll do.

Reply:
Maybe just read in normal mode and then convert to bytes? And why do we need bytes anyway? Doesn't this complicate matters?

Reply:
Sketched out [here](https://github.com/cython/cython/pull/4241/commits/d9ab66765f5e6207b60ce9ff1716d4e88d75c0f0), it remains to add tests and use

Reply:
[resolved](https://github.com/cython/cython/pull/4241/commits/05ed378b474bf6552a642eba259d0eba448bb747)